calculators/_memoria_matchups.php

```diff
@@ -20,38 +21,45 @@ function _memoria_matchups__main(&$teams) {
     if (count($teams[0]) !== count($teams[1])) {
         return;
     }
-    
+
     $game_duration_seconds = floor($teams[0][0]['duration'] / 22.4);
     if ($game_duration_seconds < 240) {
         return;
     }
-    
+
     $factor = 4000;
     $matchup_predictions = [];
     $matchup_predictions_weight = [];
     $matchup_predictions_weight_total = 0;
     for ($i = 0; $i < count($teams[0]); $i++) {
         $elo_part = abs(($teams[0][$i]['rating'] - $teams[1][$i]['rating']) / $factor);
         $matchup_predictions[$i] = floor(1000 / (1 + 10 ** $elo_part)) / 1000;
-		
+        if (count($teams[0]) > 1) {
+            $matchup_predictions[$i] -= $matchup_predictions[$i] ** (2 / (0.05 + $elo_part));
+            if ($matchup_predictions[$i] < 0) {
+                $matchup_predictions[$i] = 0;
+            }
+        }
+
         if ($teams[0][$i]['rating'] > $teams[1][$i]['rating']) {
             $matchup_predictions[$i] = 1 - $matchup_predictions[$i];
         }
-        
-        $matchup_predictions_weight[$i] = ((abs(0.5 - $matchup_predictions[$i]) + 0.015) * 6) ** ($elo_part * 1.5);
+
+        $matchup_predictions_weight[$i] = ((abs(0.5 - $matchup_predictions[$i]) + 1) * 3) ** ($elo_part / 1.25 + (max($teams[0][$i]['rating'], $teams[1][$i]['rating']) / $factor));
         $matchup_predictions_weight_total += $matchup_predictions_weight[$i];
     }
-    
+
     $prediction = 0;
     for ($i = 0; $i < count($teams[0]); $i++) {
         $normalised_weight_mult = $matchup_predictions_weight[$i] / $matchup_predictions_weight_total;
         $prediction += $matchup_predictions[$i] * $normalised_weight_mult;
     }
-    
+
+    $max_reward = 250 / max(1, max(1.25, from($teams[0])->max('$v["rating"]') / $factor, from($teams[1])->max('$v["rating"]') / $factor) - 0.25, max(1, $factor / (from($teams[0])->min('$v["rating"]') + 2500), $factor / (from($teams[1])->min('$v["rating"]') + 2500)));
```

Comment:
I'll accept it if you want, but I advise you separating this long formula into like 5 separate lines, to make it easier to read and understand.

Reply:
I mean spreading it into separate lines with assigning different variables. Like
```
$min1 = ;
$min2 = ;
$team1_coefficient = ;
$team2_coefficient = ;
$max_reward = $min1/$min2 ** ($team1_c * $team2_c)
```

if you want to do a single formula multiline like you did, following lines should be shifted right 4 spaces.